mdn/developer-portal
# 1288: Events page -- include past events

This changeset adds support for past events as an option in the Events-page sidebar filtering. It also adds pagination to the events list, which was absent from the design-focused #1266.

![Screenshot_2020-03-23 Events](https://user-images.githubusercontent.com/101457/77267850-d7888300-6c9b-11ea-8146-dfcd5a6509b3.png)

When selected, the results will be scoped to all appropriate past events (further filtered by any additional options checked, such as topic or region). If other dates are selected, matching Events will be shown, too.

This changeset also reverses the ordering of the events show, so that we now have the newest first. This is to stop the old events dominating the results when 'include past events' is selected. The order of the dates in the filter panel also reflects this.

Also, the Country list has been expanded to include all places for past events, too, else it would not be possible to filter by all relevant countries when 'include past events' is ticked. 

(Related issue #1288 )

## How to test

- The code is staged on dev. Please try out the various filter groups. eg:
  -  if ONLY 'past events' in Dates is selected, it scopes to all past events in the results, including applying any relevant filters to it (so you can find future and past "CSS" events, for instance)
  - if you filter by region or P&T and ONLY have past events selected, you ONLY get to see past events
